.github/workflows/enforce_imports.yml

```diff
@@ -0,0 +1,45 @@
+# Copyright 2024 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+name: Enforce Fully Qualified Imports
+
+on:
+  pull_request:
+    paths:
+      - '**/*.py'
+
+jobs:
+  check_imports:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ "3.9"]
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+        cache: pip
+
+    - name: Get Modified Files
+      run: |
+        BASE_BRANCH=${{ github.event.pull_request.base.ref }}
+        git diff --name-only $(git merge-base HEAD $BASE_BRANCH) > modified_files.txt
+
+    - name: Run Import Check
+      run: |
+        python automation/scripts/check_fully_qualified_imports.py $(cat modified_files.txt)
```

Comment:
We should allow such imports (e.g. `from mlrun import get_or_create_project`) in tests, docs and examples